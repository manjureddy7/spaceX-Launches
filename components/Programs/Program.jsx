import React from "react";
import Image from "next/image";
import styles from "../../styles/Program.module.css";

// const DEFAULT_IMAGE_URL =
//   "https://techcrunch.com/wp-content/uploads/2019/09/Starship-Mk1-Day.jpg";

// To improve perfomance
const DEFAULT_IMAGE_URL = "/default.webp";

const Program = ({
  name,
  id,
  missionIds,
  launchYear,
  successfulLaunch,
  successfulLanding,
  image,
}) => {
  const missionImage = image.length > 0 ? image[0] : DEFAULT_IMAGE_URL;
  const landSuccess = successfulLanding[0].land_success;
  return (
    <div className={styles.ProgramContainer}>
      <div className={styles.ImageContainer}>
        <img
          className={styles.ProgramImage}
          src={missionImage}
          alt="SPACE X MISSION IMAGE"
          loading="lazy"
          height="200px"
          width="200px"
        />
        {/* <Image
          className={styles.ProgramImage}
          src={missionImage}
          alt="SPACE X MISSION IMAGE"
          layout="fill"
        /> */}
      </div>
      <p className={styles.ProgramName}>
        {name} {` `}#{id}
      </p>
      <div className={styles.ProgramLaunchDetails}>
        <strong>Mission Ids: </strong>
        {""}
        {missionIds.length > 0 ? (
          <ul>
            {missionIds.map((missionId, index) => (
              <li key={index} className={styles.ProgramDetails}>
                {missionId}
              </li>
            ))}
          </ul>
        ) : (
          <span>N/A</span>
        )}
      </div>
      <div className={styles.ProgramLaunchDetails}>
        <strong>Launch Year:</strong>{" "}
        <span className={styles.ProgramDetails}>{launchYear}</span>
      </div>
      <div className={styles.ProgramLaunchDetails}>
        <strong>
          Successful Launch:{" "}
          <span className={styles.ProgramDetails}>
            {successfulLaunch ? "True" : "False"}
          </span>
        </strong>
      </div>
      <div className={styles.ProgramLaunchDetails}>
        <strong>
          Successful Landing:{" "}
          <span className={styles.ProgramDetails}>
            {landSuccess ? "True" : "False"}
          </span>
        </strong>
      </div>
    </div>
  );
};

export default Program;
