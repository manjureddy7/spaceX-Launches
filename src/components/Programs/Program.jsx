import React from "react";
import styles from "../../../styles/Program.module.css";

// To improve perfomance
const DEFAULT_IMAGE_URL = "/default.webp";

const Program = ({
  name,
  id,
  missionIds,
  launchYear,
  successfulLaunch,
  successfulLanding,
  image,
}) => {
  const missionImage = image.length > 0 ? image[0] : DEFAULT_IMAGE_URL;
  const landSuccess = successfulLanding[0].land_success;
  return (
    <div className={styles.ProgramContainer}>
      <div className={styles.ImageContainer}>
        <img
          className={styles.ProgramImage}
          src={missionImage}
          alt="SPACE X MISSION IMAGE"
          loading="lazy"
          height="200px"
          width="200px"
        />
      </div>
      <p className={styles.ProgramName}>
        {name} {` `}#{id}
      </p>
      <div className={styles.ProgramLaunchDetails}>
        <strong>Mission Ids: </strong>
        {""}
        {missionIds.length > 0 ? (
          <ul>
            {missionIds.map((missionId, index) => (
              <li key={index} className={styles.ProgramDetails}>
                {missionId}
              </li>
            ))}
          </ul>
        ) : (
          <span>N/A</span>
        )}
      </div>
      <div className={styles.ProgramLaunchDetails}>
        <strong>Launch Year:</strong>{" "}
        <span className={styles.ProgramDetails}>{launchYear}</span>
      </div>
      <div className={styles.ProgramLaunchDetails}>
        <strong>
          Successful Launch:{" "}
          <span className={styles.ProgramDetails}>
            {successfulLaunch ? "True" : "False"}
          </span>
        </strong>
      </div>
      <div className={styles.ProgramLaunchDetails}>
        <strong>
          Successful Landing:{" "}
          <span className={styles.ProgramDetails}>
            {landSuccess ? "True" : "False"}
          </span>
        </strong>
      </div>
    </div>
  );
};

export default Program;
