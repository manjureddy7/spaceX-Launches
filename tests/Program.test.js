import React from "react";
import { render, cleanup, screen } from "@testing-library/react";

import Program from "../components/Programs/Program";

afterEach(cleanup);

const props = {
  name: "Test Rocket",
  id: "",
  missionIds: ["212"],
  launchYear: 2006,
  successfulLaunch: true,
  successfulLanding: [{ land_success: false }],
  image: [""],
};

describe("Program Component", () => {
  test("Should match snapshot", () => {
    const { asFragment } = render(<Program {...props} />);
    expect(asFragment(<Program />)).toMatchSnapshot();
  });
  test("renders Program component", () => {
    const { getByText } = render(<Program {...props} />);
    expect(getByText("Launch Year:")).toBeInTheDocument();
    expect(getByText("Successful Launch:")).toBeInTheDocument();
    expect(getByText("Successful Landing:")).toBeInTheDocument();
  });
  test("Test elements", () => {
    render(<Program {...props} />);
    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(screen.getByAltText("SPACE X MISSION IMAGE")).toBeInTheDocument();
  });
});

// Refer these docs in future for Complete Testing Library

// https://www.robinwieruch.de/react-testing-library
// https://www.freecodecamp.org/news/8-simple-steps-to-start-testing-react-apps-using-react-testing-library-and-jest/
