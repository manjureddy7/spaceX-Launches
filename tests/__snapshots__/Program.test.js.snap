// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Program Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProgramContainer"
  >
    <div
      class="ImageContainer"
    >
      <img
        alt="SPACE X MISSION IMAGE"
        class="ProgramImage"
        loading="lazy"
        src=""
      />
    </div>
    <p
      class="ProgramName"
    >
      Test Rocket  #
    </p>
    <div
      class="ProgramLaunchDetails"
    >
      <strong>
        Mission Ids: 
      </strong>
      <ul>
        <li
          class="ProgramDetails"
        >
          212
        </li>
      </ul>
    </div>
    <div
      class="ProgramLaunchDetails"
    >
      <strong>
        Launch Year:
      </strong>
       
      <span
        class="ProgramDetails"
      >
        2006
      </span>
    </div>
    <div
      class="ProgramLaunchDetails"
    >
      <strong>
        Successful Launch: 
        <span
          class="ProgramDetails"
        >
          True
        </span>
      </strong>
    </div>
    <div
      class="ProgramLaunchDetails"
    >
      <strong>
        Successful Landing: 
        <span
          class="ProgramDetails"
        >
          False
        </span>
      </strong>
    </div>
  </div>
</DocumentFragment>
`;
